### ISSUE TITLE
Test 3.7-C - is specific to the modeshape implementation, not API spec

### ISSUE THREAD

**Issue by zimeon** [fcrepo4-labs/Fedora-API-Test-Suite#94]
The test 3.7-C, checking for illegal attempts to modify server managed triples:

https://github.com/fcrepo4-labs/Fedora-API-Test-Suite/blob/c30f47aa55c89ed1336cf4944aff104ce5e6bfa5/src/main/java/org/fcrepo/spec/testsuite/crud/HttpPatch.java#L141-L171

with SPARQL update code:

https://github.com/fcrepo4-labs/Fedora-API-Test-Suite/blob/c30f47aa55c89ed1336cf4944aff104ce5e6bfa5/src/main/java/org/fcrepo/spec/testsuite/crud/HttpPatch.java#L53-L57

is checking for rejection of a change to `fedora:lastModifiedBy` which is not defined in the API specification (or LDP of course).

Since there is no mechanism in the API specification to find out what special triples a server manages I think this test has to be removed from the test suite. I wonder whether there should be an extra collection of tests kept aside for the modeshape fedora 5?

**Comment by awoods** [comment]
The essence of this requirement is that if the client performs a PATCH and the implementation is unable to satisfy that request, it should fail with a 409, [otherwise must return a 2xx response].

Would it make sense to have tests that exercise scenarios that fail against existing implementations, and if the response is 2xx, that passes the test as well?

**Comment by zimeon** [comment]
Well, that would provide some test but would be very weak. A slightly stronger strategy I had thought of is the idea of creating an empty LDPRS, then trying to change the triples that appear when you GET it, on the assumption that these are server managed?

**Comment by awoods** [comment]
Sounds like a plan.

**Comment by dbernstein** [comment]
It seems the suggested approach will leave the functionality untested for implementations without server-managed triples (trellis).  @acoburn suggested simply trying to insert a triple disallowed by LDP such as <> rdf:type "a literal" since rdf:types must take a URI as an object. Any objections to this approach?

**Comment by acoburn** [comment]
Just to clarify: LDP doesn't explicitly disallow the triple pattern `<> rdf:type "a literal"`, though that triple doesn't make much semantic sense. This is just an example of a triple pattern that an implementation might disallow (Trellis disallows that).

**Comment by zimeon** [comment]
LDP does not require testing of RDF Schema validity so I don't see why a server is required to reject `<> rdf:type "A thingy!".`

**Comment by zimeon** [comment]
Oops... I see a just repeated what @acoburn [said](https://github.com/fcrepo/Fedora-API-Test-Suite/issues/94#issuecomment-423667781)

**Comment by acoburn** [comment]
I completely agree with @zimeon. A better test would be for something that LDP explicitly disallows, like setting both `ldp:hasMemberRelation` *and* `ldp:isMemberOfRelation` on a Direct/Indirect resource.
